From 54f907393560090d22307fe02fefe7c5e1715e94 Mon Sep 17 00:00:00 2001
From: Xianzhong <b07117@freescale.com>
Date: Tue, 12 Aug 2014 01:50:25 +0800
Subject: [PATCH] ENGR00326593 [#1297] fix virtual memory database query

it is not complete in the original implementation to query virtual command buffer,
it is necessary to this fix to get the correct GPU virtual memory result.
also include virtual command buffer database for Vivante gcDB tool.

Date: Aug 11, 2014
Signed-off-by: Xianzhong <b07117@freescale.com>
Acked-by: Jason Liu
(cherry picked from commit 81fe8a98067132ba830be5220d554c7920b729ac)
(cherry picked from commit 4899c9881e611fdd0bedf59b5d657c521ee92e88)
---
 drivers/mxc/gpu-viv/hal/kernel/gc_hal_kernel.c | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/drivers/mxc/gpu-viv/hal/kernel/gc_hal_kernel.c b/drivers/mxc/gpu-viv/hal/kernel/gc_hal_kernel.c
index 34a8979..15b518d 100644
--- a/drivers/mxc/gpu-viv/hal/kernel/gc_hal_kernel.c
+++ b/drivers/mxc/gpu-viv/hal/kernel/gc_hal_kernel.c
@@ -1246,6 +1246,7 @@ gckKERNEL_QueryDatabase(
 {
     gceSTATUS status;
     gctINT i;
+    gcuDATABASE_INFO tmp;
 
     gceDATABASE_TYPE type[3] = {
         gcvDB_VIDEO_MEMORY | (gcvPOOL_SYSTEM << gcdDB_VIDEO_MEMORY_POOL_SHIFT),
@@ -1303,7 +1304,15 @@ gckKERNEL_QueryDatabase(
                                  Interface->u.Database.processID,
                                  !Interface->u.Database.validProcessID,
                                  gcvDB_COMMAND_BUFFER,
-                                 &Interface->u.Database.vidMemPool[2]));
+                                 &tmp));
+
+    Interface->u.Database.vidMemPool[2].counters.bytes += tmp.counters.bytes;
+    Interface->u.Database.vidMemPool[2].counters.maxBytes += tmp.counters.maxBytes;
+    Interface->u.Database.vidMemPool[2].counters.totalBytes += tmp.counters.totalBytes;
+
+    Interface->u.Database.vidMem.counters.bytes += tmp.counters.bytes;
+    Interface->u.Database.vidMem.counters.maxBytes += tmp.counters.maxBytes;
+    Interface->u.Database.vidMem.counters.totalBytes += tmp.counters.totalBytes;
 
 #if gcmIS_DEBUG(gcdDEBUG_TRACE)
     gckKERNEL_DumpVidMemUsage(Kernel, Interface->u.Database.processID);
-- 
1.8.0

